pallets/asset/src/lib.rs

```diff
@@ -1854,17 +1877,27 @@ impl<T: Config> Module<T> {
         Ok(())
     }
 
-    fn base_redeem(origin: T::Origin, ticker: Ticker, value: Balance) -> DispatchResult {
+    fn base_redeem(
+        origin: T::Origin,
+        ticker: Ticker,
+        value: Balance,
+        portfolio_kind: PortfolioKind,
+    ) -> DispatchResult {
         // Ensure origin is agent with custody and permissions for default portfolio.
-        let agent = Self::ensure_agent_with_custody_and_perms(origin, ticker)?;
+        let agent = Self::ensure_agent_with_custody_and_perms(origin.clone(), ticker)?;
```

Comment:
This clone shouldn't be needed.
```suggestion
        let agent = Self::ensure_agent_with_custody_and_perms(origin, ticker)?;
```